### PR TITLE
Port Orgs to ListRow&co

### PR DESCRIPTION
**Pull request by hawkrives**
> This is a PR in a series of conversion PRs, split out of #499

The biggest "huh?" in this PR is likely this line:

```js
rowHeight + (Platform.OS === 'ios' ? (11/12 * StyleSheet.hairlineWidth) : 0)
```

11/12ths of hairlineWidth?

You might have noticed the way that the iOS AlphabetListViews never quite scroll to the right spot. With 11/12ths of hairlineWidth, they do (nearly). 12/12ths is too much.

| Platform | Before | After |
| ---:| --- | --- |
| iOS | ios orgs a | ios orgs b
| Android | android orgs a | android orgs b

No separators on this one, because the rows are two lines long.

### CODE REVIEW

File: views/student-orgs/list.js [hunk 1]
```diff
@@ -181,28 +156,18 @@ export class StudentOrgsView extends React.Component {
...
-          justifyContent: 'center',
-          alignItems: 'center',
-          backgroundColor: '#ffffff',
-        }}>
-          <Text>
-            No organizations found.
-          </Text>
-        </View>
-      )
+      return <NoticeView text='No organizations found.' />
```
Comment (hawkrives) [inline views/student-orgs/list.js:159]: NoticeView to the rescue again!

File: views/student-orgs/list.js [hunk 2]
```diff
@@ -181,28 +156,18 @@ export class StudentOrgsView extends React.Component {
...
         data={this.state.orgs}
         cell={this.renderRow}
-        sectionHeader={this.renderHeader}
-        sectionHeaderHeight={28}
-        cellHeight={53}
+        // just setting cellHeight sends the wrong values on iOS.
+        cellHeight={rowHeight + (Platform.OS === 'ios' ? (11/12 * StyleSheet.hairlineWidth) : 0)}
+        sectionHeader={this.renderSectionHeader}
+        sectionHeaderHeight={headerHeight}
+        renderSeparator={this.renderSeparator}
```
Comment (hawkrives) [inline views/student-orgs/list.js:170]: Turns out that you can use renderSeparator on AlphabetListView, too. That simplified some code!

File: views/student-orgs/list.js [hunk 3]
```diff
@@ -181,28 +156,18 @@ export class StudentOrgsView extends React.Component {
...
-      <AlphabetListView
-        contentContainerStyle={styles.listView}
+      <StyledAlphabetListView
         data={this.state.orgs}
         cell={this.renderRow}
-        sectionHeader={this.renderHeader}
-        sectionHeaderHeight={28}
-        cellHeight={53}
+        // just setting cellHeight sends the wrong values on iOS.
+        cellHeight={rowHeight + (Platform.OS === 'ios' ? (11/12 * StyleSheet.hairlineWidth) : 0)}
```
Comment (hawkrives) [inline views/student-orgs/list.js:167]: hello magic numbers

File: views/student-orgs/list.js [hunk 4]
```diff
@@ -91,12 +51,12 @@ export class StudentOrgsView extends React.Component {
   }
 
   state: {
-    orgs: StudentOrgAbridgedType[],
+    orgs: {[key: string]: StudentOrgAbridgedType[]},
```
Comment (hawkrives) [inline views/student-orgs/list.js:54]: The old type was inaccurate. I didn't change how they're stored, I just fixed the type. I'm not sure why Flow didn't complain before.

**Review by drewvolz — APPROVED** [review]

**Pull request merged**